strax/corrections/cmt_DB_interface.py

```diff
@@ -0,0 +1,115 @@
+
+        self.host = host
+        self.username = username
+        self.password = password
+        self.client = pymongo.MongoClient(host=self.host,username=self.username,password=self.password)
+        self.DATABASE_NAME= 'corrections' 
+        self.DATABASE=self.client[self.DATABASE_NAME]
+
+    def list_corrections(self):
+        """Smart logic to list corrections
+        """
+        return [x['name'] for x in self.DATABASE.list_collections() if not 'global' in x['name']]
+
+
+    def read(self, correction):
+        """Smart logic to read corrections
+        """ 
+        df = pdm.read_mongo(correction, [], self.DATABASE)
+    
+        # No data found
+        if df.size == 0:
+            return None
+        # Delete internal Mongo identifier
+        del df['_id']
+
+        # Set UTC
+        #df['time'] = pd.to_datetime(df['time'], utc=True)
+    
+        return  df.set_index('time')
+
+    def interpolate(self, what, when, how='interpolate'):
+        df = what
+    
+        df_new = pd.DataFrame.from_dict({'Time' : [when],
+                                    # 'This' : [True]
+                                    })
+        df_new = df_new.set_index('Time')
+    
+        df_combined = pd.concat([df,
+                             df_new],
+                            sort=False)
+        df_combined = df_combined.sort_index()
+        if how == 'interpolate':
+            df_combined = df_combined.interpolate(method='linear')
+        elif how == 'fill':
+            df_combined = df_combined.ffill()
+        else:
+            raise ValueError()
+        
+        return df_combined
+
+    def get_context_config(self, when, global_version = 'v1', xenon1t=False):
+        """Global configuration logic
+        """ 
+        if xenon1t:
+            df_global = Corrections_DB_interface.read(self,'global_xenon1t')
+        else:
+            df_global = Corrections_DB_interface.read(self,'global')
+    
+        context_config = {}
+    
+        for correction, version in df_global.iloc[-1][global_version].items():
+            df = Corrections_DB_interface.read(self,correction)
+
+            df = Corrections_DB_interface.interpolate(self,df, when)
+            context_config[correction] = df.loc[df.index == when, version].values[0]
+    
+        return context_config
+
+    def write(self, correction, df):
+        """Smart logic to write corrections
+        """
+        if 'ONLINE' not in df.columns:
+            raise ValueError('Must specify ONLINE column')
+        if 'v1' not in df.columns:
+            raise ValueError('Must specify v1 column')
+       
+        # Set UTC
+        #df.index = df.index.tz_localize('UTC')
+    
+        # Compare against
+        logging.info('Reading old values for comparison')
+        df2 = Corrections_DB_interface.read(self,correction)
+    
+        if df2 is not None:
+            logging.info('Checking if columns unchanged in past')
+
+            now = datetime.now()
```

Comment:
Careful with such things this will return your local time I am note sure if this is what you want.

Reply:
you can use ```datetime.now(pytz.utc)```

Reply:
fixed